### PR TITLE
Enable `Get-ChildItem` to produce `Mode` property even if cannot determine if hardlink

### LINKED ISSUE #7353: PowerShell v6.1.0-preview.4: Broken output for Get-ChildItem
Steps to reproduce
------------------
Simply run `ls -Force` in your usual _Home_-directory (i.e. `C:\Users\Hrxn`, for example)...
```powershell
ls -fo
```

Expected behavior
-----------------
Output (snipped)
```powershell
    Directory: C:\Users\Hrxn


Mode                LastWriteTime         Length Name                                                                                                                                                  
----                -------------         ------ ----  
[....]
-a----       29.11.2015     15:39            265 lynx.cmd                                                                                                                                              
-a-h--       25.07.2018     01:59       16515072 NTUSER.DAT                                                                                                                                            
-a-hs-       03.05.2018     06:57        2949120 ntuser.dat.LOG1                                                                                                                                       
-a-hs-       03.05.2018     06:57        3145728 ntuser.dat.LOG2                                                                                                                                       
-a-hs-       03.05.2018     06:57          65536 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TM.blf                                                                                               
-a-hs-       03.05.2018     06:57         524288 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TMContainer00000000000000000001.regtrans-ms                                                          
-a-hs-       03.05.2018     06:57         524288 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TMContainer00000000000000000002.regtrans-ms                                                          
---hs-       03.05.2018     07:03             20 ntuser.ini                                                                                                                                            
-arhs-       03.05.2018     07:03            426 ntuser.pol                                                                                                                                            
-a----       30.09.2016     20:26            821 SciTEUser.properties
[....]
```

Actual behavior
---------------
Output (snipped)
```powershell
    Directory: C:\Users\Hrxn


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
[....]
-a----       29.11.2015     15:39            265 lynx.cmd
             25.07.2018     01:59       16515072 NTUSER.DAT
             03.05.2018     06:57        2949120 ntuser.dat.LOG1
             03.05.2018     06:57        3145728 ntuser.dat.LOG2
             03.05.2018     06:57          65536 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TM.blf
             03.05.2018     06:57         524288 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TMContainer00000000000000000001.regtrans-ms
             03.05.2018     06:57         524288 NTUSER.DAT{91ed9ba6-4e96-11e8-a51f-00224d878c36}.TMContainer00000000000000000002.regtrans-ms
---hs-       03.05.2018     07:03             20 ntuser.ini
-arhs-       03.05.2018     07:03            426 ntuser.pol
-a----       30.09.2016     20:26            821 SciTEUser.properties
[....]
```

As you can probably see, the output is somewhat garbled. 
The Mode/Attributes for some files are missing, when it should be `-a-hs-`. Seems like some rendering issue.
Running v6.1.0-preview.4 in Administrator Mode exhibits the same behavior.

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.4
PSEdition                      Core
GitCommitId                    6.1.0-preview.4
OS                             Microsoft Windows 10.0.17134 
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

NB: The _Expected Behavior_ / Working Output was done with the OS Powershell (`PSVersion                      5.1.17134.165`)

OS info:
```
Windows 10 Pro
1803

OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.17134 N/A Build 17134
```

Would be interesting to know if someone can reproduce...

PS:
I don't even know what these files are.. probably not important anymore?

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Some files in the user directory on Windows such as 'NTUSER.DAT' cause the IsHardLink() check to fail with an INVALID_HANDLE win32 error.  This causes the `Mode` property to fail being defined even though it was fine up until that last api call.  I don't see any reason to throw an exception if the hard link check fails, instead just return false since we cannot determine it's a hardlink.

Fix https://github.com/PowerShell/PowerShell/issues/7353

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by kvprasoon** [comment]
@SteveL-MSFT  What will be the output after this change, will it exclude such files ?

**Review by daxian-dbw — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@kvprasoon no.  With this change, it should work the same as with Windows PowerShell before we added the check for hard links.

**Comment by daxian-dbw** [comment]
@kvprasoon It will not exclude such files. The mode for such files will not have `l` in it when the call to `GetFileInformationByHandle` fails, that's all.

**Comment by kvprasoon** [comment]
@daxian-dbw Thanks for the clarification.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -8136,15 +8136,12 @@ internal static bool WinIsHardLink(ref IntPtr handle)
...
-            if (handleInfo.NumberOfLinks > 1)
-            {
-                return true;
+                if (handleInfo.NumberOfLinks > 1)
+                {
+                    return true;
+                }
             }
 
             return false;
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: We could remove if-s:
```c#
bool succeeded = InternalSymbolicLinkLinkCodeMethods.GetFileInformationByHandle(handle, out handleInfo);

return succeeded && (handleInfo.NumberOfLinks > 1);
```

**Review by iSazonov — APPROVED** [review]
LGTM with one minor comment.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Good suggestion.  Will update.

**Comment by TravisEz13** [comment]
@daxian-dbw FYI, I had to go back to the PR to figure this one out.  The message used to commit wasn't as good as the PR title

**Pull request merged**